lib/index.js

```diff
@@ -798,6 +798,12 @@ function createExpression(source) {
     expression = (path) ?
       createPathExpression(path) :
       new expressions.Expression();
+    if (meta.as) {
```

Comment:
Perhaps we should just declare a variable above when we get the value from the regex and then only assign the value to meta that is returned from parsing it.

Up above we can do a:
```
as = match[3];
```

and then down here we can do
```
if (as) {
  meta.as = parseAlias(as);
}
```

That way we are sure that the way the alias is parsed is consistent in the definition and where it is used within paths later.